### ISSUE TITLE
train tiny-yolo..again..

### ISSUE THREAD

**Issue by RushNuts** [pjreddie/darknet#581]
hi there.

how i can train using yolov2-tiny.cfg (coco set) train tiny weights?
_Where i get pre-trained model?_ 

yolov2-tiny.cfg and tiny-yolo.cfg Is different configs. thanks!

**Comment by TheMikeyR** [comment]
I guess tiny-yolo.cfg is leftover from earlier commit (you probably did some changes). It got renamed to yolov2-tiny.cfg. There are no changes in the two version just rename as you can see [here](https://www.diffchecker.com/GAkEAyQ0) tiny-yolo.cfg compared to yolov2-tiny.cfg the different configs are probably due to your own modifications. 

Pre-trained model available at https://pjreddie.com/darknet/yolo/ in the table

**Comment by RushNuts** [comment]
this already trained model for detection. I need a model for learning my data

**Comment by TheMikeyR** [comment]
Then take [this](https://pjreddie.com/media/files/yolov2-tiny-voc.weights) weightfile and use `./darknet partial tiny-yolo-voc.cfg tiny-yolo-voc.weights tiny-yolo-voc.conv.13 13` and adapt it to your cfg. 
The number 13 tells how many layer you want, you want e.g. just before the final layer which does the detection. 

If the command doesn't exist in pjreddies repo then you can use alexeyab's repo instead. In Alexey's repo you are also able to train without any initialized weights, by issuing the train command without any weights (then it will initialize with random weights).

**Comment by RushNuts** [comment]
I wanna train with coco data, unfortunately in your example for VOC :( but thanks

**Comment by TheMikeyR** [comment]
@RushNuts the syntax is the same for coco it is just an example.

**Comment by RushNuts** [comment]
@TheMikeyR Hi again. look, how i can download the last version of Yolo v2? Which branch?

**Comment by TheMikeyR** [comment]
yolov2 is the same in both repositories, the cfg isn't changed it either. You can get the weights from https://pjreddie.com/darknet/yolo/ or use the old site dedicated for yolov2 https://pjreddie.com/darknet/yolov2/

**Comment by MizbaMohammed** [comment]
@TheMikeyR how can I train my tiny YOLO for some objects from COCO dataset?

**Comment by TheMikeyR** [comment]
@MizbaMohammed https://pjreddie.com/darknet/yolov2/ follow the section "Training YOLO on COCO" adapt the files to the classes you want to train from coco, and update the lines to use tiny-yolo. You should also follow "Download Pretrained Convolutional Weights" and download the tiny-yolo weights and extract the 13 first layers.

**Comment by MizbaMohammed** [comment]
@TheMikeyR  Hey Mike, I want to train the tiny YOLO which was trained on VOC dataset to train for few objects in COCO. I just have to put the tiny YOLO VOC instead of the pretrained conv weights right?

**Comment by MizbaMohammed** [comment]
@TheMikeyR what all changes do I have to make in that case?

**Comment by TheMikeyR** [comment]
@MizbaMohammed try Alexey repo https://github.com/AlexeyAB/darknet
Here is the code https://github.com/AlexeyAB/darknet/blob/master/src/detector.c#L839

**Comment by TheMikeyR** [comment]
You need to change the filtersize in the config to present the amount of classes you want to train, info on this in the website or multiple issue threads.

**Comment by abdulkalam1233** [comment]
can anybody explain this
@pjreddie 
[convolutional]	
size=1	
stride=1	
pad=1	
filters=425	
activation=linear

why batch_norm is not added and why activation is linear?

**Comment by keides2** [comment]
@TheMikeyR Where can I download tiny-yolo-voc.conv.13?

**Comment by TheMikeyR** [comment]
https://github.com/pjreddie/darknet/issues/581#issuecomment-376825207

**Comment by abdulkalam1233** [comment]
@TheMikeyR 
man i have  a doubt  can i label both dog and cat for annotation in only one txt file while training in below image 
![fotolia_8527249_xs](https://user-images.githubusercontent.com/39231568/42144885-400bc18c-7ddb-11e8-93e0-857d6ed0d246.jpg)

**Comment by TheMikeyR** [comment]
@abdulkalam1233 yes and then you need to put a bounding box with the two classes around cat and dog. 

Check out the papers
https://pjreddie.com/media/files/papers/yolo_1.pdf
https://pjreddie.com/media/files/papers/YOLO9000.pdf
https://pjreddie.com/media/files/papers/YOLOv3.pdf

**Comment by abdulkalam1233** [comment]
@TheMikeyR 
Can you share one of your dataset that you used for training?

**Comment by TheMikeyR** [comment]
@abdulkalam1233 let me refer to this excellent article on how to train yolo on a custom dataset https://timebutt.github.io/static/how-to-train-yolov2-to-detect-custom-objects/
For multi classes dataset, this helped me getting started http://guanghan.info/blog/en/my-works/train-yolo/

**Comment by abdulkalam1233** [comment]
@TheMikeyR 
Thank you

**Comment by abdulkalam1233** [comment]
@TheMikeyR 
how can I add an extra dataset to the final weight file?
means how to train final weight file without losing the previous weights?

**Comment by gustavovaliati** [comment]
Hi @abdulkalam1233 ,
I suppose one valid approach would be to mix both datasets in the training process.
Maybe that would be good for better generalization power when testing in a third dataset. At the same time, I also suppose that testing that model in just one of the trained datasets would provide worse score compared to the model trained specifically for that dataset.
In summary, you would have a generalization gain but would have a loss for those specific datasets.

That is what I think. If you have tested or will test that, would be nice to share the results with us ;D

**Comment by jalaldev1980** [comment]
Tiny YOLO V2:

I explained on https://github.com/dwaithe/yolov2 the pre trained weight tiny can be generated using 

darknet.exe partial cfg/tiny-yolo-voc.cfg tiny-yolo-voc.weights tiny-yolo-voc.conv.13 13


what is the “13” if I am training 5 objects do I need to change it ?

do I need to use my cfg file instead of tiny-yolo-voc.cfg?

thank you in advance

**Comment by beizhengren** [comment]
> If the command doesn't exist in pjreddies repo then you can use alexeyab's repo instead. In Alexey's repo you are also able to train without any initialized weights, by issuing the train command without any weights (then it will initialize with random weights).

@TheMikeyR Hello, you mentioned in Alexey's repo can train without any initialized weights, why?
What's the effect between above and  with parital weights in pjreddies repo ?
And you said `it will initialize with random weights`, does pjreddies repo have a  initialize with random weights, if traing without any weights?
TheMikeyR, Thanks
